internal/server/tls.go

```diff
@@ -17,20 +16,50 @@ import (
 	"github.com/infrahq/infra/internal/logging"
 )
 
+// MapCache is a simple in-memory caching mechanism, it is not thread safe
+type MapCache map[string][]byte
+
+func (m MapCache) Get(ctx context.Context, name string) ([]byte, error) {
+	data, cached := m[name]
+	if !cached {
+		return nil, autocert.ErrCacheMiss
```

Comment:
I think we can use our own error, since we aren't using this with `autocert.Manager` anymore. Could also be a second pass cleanup.


```suggestion
		return nil, fmt.Errorf("no cache entry for %v", name)
```

Reply:
Gonna leave this for now because I know that getCertificate is called within the cert manager, so Im worried they rely on this error type